micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java

```diff
@@ -250,4 +264,91 @@ private SessionFactory unwrap(EntityManagerFactory entityManagerFactory) {
         }
     }
 
+    /**
+     * Register hibernate loadEvent listener - event trigger the metrics registration,
+     * only select queries are recorded in QueryStatistics {@link org.hibernate.stat.spi.StatisticsImplementor}
+     * @param meterRegistry meterRegistry to use
+     */
+    private void registerListeners(MeterRegistry meterRegistry) {
+        if (Objects.nonNull(sessionFactory) && sessionFactory instanceof SessionFactoryImplementor) {
+            EventListenerRegistry registry = ((SessionFactoryImplementor) sessionFactory).getServiceRegistry().getService(EventListenerRegistry.class);
+            MetricsEventHandler metricsEventHandler = new MetricsEventHandler(meterRegistry);
+            registry.appendListeners(EventType.POST_LOAD, metricsEventHandler);
+        }
+
+    }
+
+    class MetricsEventHandler implements PostLoadEventListener {
+
+        private final MeterRegistry meterRegistry;
+
+        MetricsEventHandler(MeterRegistry meterRegistry) {
+            this.meterRegistry = meterRegistry;
+        }
+
+        @Override
+        public void onPostLoad(PostLoadEvent event) {
+            registerQueryMetric(event.getSession().getFactory().getStatistics());
+        }
+
+        void registerQueryMetric(Statistics statistics) {
+            for (String query : statistics.getQueries()) {
+                QueryStatistics queryStatistics = statistics.getQueryStatistics(query);
+                if (Objects.nonNull(queryStatistics)) {
+                    String queryName = query.replace(" ", "_").replace(".", "_").toLowerCase();
```

Comment:
What kind of cardinality can be expected for this? I suppose it depends on whether bind variables are used and if the query is somehow normalized.

Reply:
You're asking for example? It can be:
`Select generatedAlias0 from Table as generatedAlias0 where generatedAlias0.param0 :val0`

Reply:
I mean how many unique values for this queryName can we expect in applications that use Hibernate? I'm thinking it might be a lot and if that is the case, maybe this should be an opt-in feature that can be turned on for those that want to track query-level metrics.

Reply:
@shakuzen I think that it would make sense to make it opt-in feature, since number of these metrics can be quite high

Reply:
@shakuzen Do you have any strategy for making a metrics configurable? So far I've seen that all are enabled by default without any additional configuration

Reply:
Adding a boolean flag to the constructor would be one way, but there are already a lot of arguments in the constructor and this doesn't scale if we wanted more opt-in metrics, so I don't think we should go this route. Making it a separate `MeterBinder` is probably the easiest way to make it not automatically included with the current `MeterBinder`.

Reply:
Moved it to separate MeterBinder